packages/utils/src/severity.ts

```diff
@@ -1,20 +1,28 @@
-import { Severity } from '@sentry/types';
+import { Severity, SeverityLevel } from '@sentry/types';
 
-import { SeverityLevel, SeverityLevels } from './enums';
+// TODO: Should `severityFromString` be deprecated?
+
+// Note: Ideally the `SeverityLevel` type would be derived from `validSeverityLevels`, but that would mean either
+//
+// a) moving `validSeverityLevels` to `@sentry/types`,
+// b) moving the`SeverityLevel` type here, or
+// c) importing `validSeverityLevels` from here into `@sentry/types`.
+//
+// Option A would make `@sentry/types` a runtime dependency of `@sentry/utils` (not good), and options B and C would
+// create a circular dependency between `@sentry/types` and `@sentry/utils` (also not good). So a TODO accompanying the
+// type, reminding anyone who changes it to change this list also, will have to do.
+
+export const validSeverityLevels = ['fatal', 'error', 'warning', 'log', 'info', 'debug', 'critical'];
 
-function isSupportedSeverity(level: string): level is Severity {
-  return SeverityLevels.indexOf(level as SeverityLevel) !== -1;
-}
 /**
- * Converts a string-based level into a {@link Severity}.
+ * Converts a string-based level into a member of the {@link Severity} enum.
  *
- * @param level string representation of Severity
+ * @param level String representation of Severity
  * @returns Severity
  */
 export function severityFromString(level: SeverityLevel | string): Severity {
-  if (level === 'warn') return Severity.Warning;
-  if (isSupportedSeverity(level)) {
-    return level;
-  }
-  return Severity.Log;
+  return (level === 'warn' ? 'warning' : validSeverityLevels.includes(level) ? level : 'log') as Severity;
```

Comment:
More of a question than a remark: Do we have a convention regarding nested ternary operators? I personally don't mind them (in fact I really like using ternary operators) but I know that this is sort of a debated topic amongst people...

Reply:
We do not. I think it's not that hard to get carried away with them (see below), so I try to use them fairly sparingly, but sometimes, especially for short stuff, I think they're fine.

_Real code; not ours_
![image](https://user-images.githubusercontent.com/14812505/163131965-14b830f9-7dc9-46d2-a7c2-db1d7c8d8092.png)